onadata/apps/viewer/models/parsed_instance.py

```diff
@@ -61,6 +61,13 @@ def dict_for_mongo(d):
         if _is_invalid_for_mongo(key):
             del d[key]
             d[_encode_for_mongo(key)] = value
+        # We want to restore nested objects. We don't want to replace keys which start or end with __
+        elif key.count("__") and not (key.startswith("__") or key.endswith("__")) > 0:
+            del d[key]
```

Comment:
I'd be kind of surprised if this code ever gets to run in Python 3, but let's change https://github.com/kobotoolbox/kobocat/pull/402/files#diff-fb6a36fb2e5974841a164dc4dff6651eR49 to `list(d.items())` while we're here to avoid weirdness since:
```
Python 3.4.3 (default, Nov 28 2017, 16:41:13) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> d = dict(zip('qwertyuiop', '1234567890'))
>>> ditems = d.items()
>>> ditemslist = list(d.items())
>>> ditems
dict_items([('q', '1'), ('t', '5'), ('w', '2'), ('u', '7'), ('i', '8'), ('e', '3'),
    ('o', '9'), ('p', '0'), ('r', '4'), ('y', '6')])
>>> ditemslist
[('q', '1'), ('t', '5'), ('w', '2'), ('u', '7'), ('i', '8'), ('e', '3'),
    ('o', '9'), ('p', '0'), ('r', '4'), ('y', '6')]
>>> del d['y']
>>> ditems
dict_items([('q', '1'), ('t', '5'), ('w', '2'), ('u', '7'), ('i', '8'), ('e', '3'),
    ('o', '9'), ('p', '0'), ('r', '4')])
>>> ditemslist
[('q', '1'), ('t', '5'), ('w', '2'), ('u', '7'), ('i', '8'), ('e', '3'),
    ('o', '9'), ('p', '0'), ('r', '4'), ('y', '6')]
```

Reply:
@jnm 
if should not be `key.count("__") and not (key.startswith("__") or key.endswith("__")) > 0:` 
but only `key.count("__") and not (key.startswith("__") or key.endswith("__"))` anyway.